common/scripts/lib/_parseArgs.sh

```diff
@@ -308,21 +308,20 @@ __prompt_for_inputs() {
 }
 
 __set_host_services() {
-  sed -i '/^HOST_SERVICES/d' "$ADMIRAL_ENV"
-  echo "HOST_SERVICES=true" >> "$ADMIRAL_ENV"
-  source "$ADMIRAL_ENV"
-
-  ## add ssh key to host's authorized keys
-  local authorized_keys="/root/.ssh/authorized_keys"
-  local tmp_authorized_keys="/root/.ssh/tmp"
-  local ssh_public_key=$(cat "$SSH_PUBLIC_KEY")
-  # remove public key from authorized_keys, if present
-  if grep -v "^$ssh_public_key" "$authorized_keys" > "$tmp_authorized_keys"; then
-    cat "$tmp_authorized_keys" > "$authorized_keys"
+  if [ "$HOST_SERVICES" == "true" ]; then
+    sed -i '/^HOST_SERVICES/d' "$ADMIRAL_ENV"
+    echo "HOST_SERVICES=true" >> "$ADMIRAL_ENV"
+    source "$ADMIRAL_ENV"
+
+    # create .ssh dir, if not present
+    local root_ssh_dir="/root/.ssh"
+    mkdir -p "$root_ssh_dir"
+
+    ## add public ssh key to host's authorized keys
+    local authorized_keys="$root_ssh_dir/authorized_keys"
+    local ssh_public_key=$(cat "$SSH_PUBLIC_KEY")
+    echo "$ssh_public_key" | tee -a "$authorized_keys" &> /dev/null
```

Comment:
We add ssh key [this way](https://github.com/Shippable/admiral/blob/master/common/scripts/x86_64/Ubuntu_14.04/_helpers.sh#L393), so following the pattern